manifests/git_repo.pp

```diff
@@ -0,0 +1,22 @@
+# Setup and configure a git repo
+define katello_devel::git_repo(
+  $source,
+  $github_username = undef,
+) {
+
+  validate_string($source)
+
+  $sources = {origin => "https://github.com/${source}.git"}
+
+  if $github_username != undef {
+    $sources[$github_username] = "https://github.com/${github_username}/${title}.git"
```

Comment:
This is probably up for debate as to whats the default for user forks: ssh or https